paasta_tools/setup_marathon_job.py

```diff
@@ -168,21 +169,27 @@ def drain_tasks_and_find_tasks_to_kill(
                 line='%s bounce draining %d old tasks with app_id %s' %
                 (bounce_method, len(tasks), app_id),
             )
-        for task, client in tasks_to_drain:
+
+        async def drain_and_kill_if_draining_fails(task: MarathonTask, client: MarathonClient) -> None:
             all_draining_tasks.add((task, client))
             if task.state == 'TASK_UNREACHABLE':
-                continue
+                return
             try:
-                drain_method.drain(task)
+                await drain_method.drain(task)
             except Exception as e:
                 log_bounce_action(
                     line=("%s bounce killing task %s due to exception when draining: %s" % (bounce_method, task.id, e)),
                 )
                 tasks_to_kill.add((task, client))
 
-    for task, client in all_draining_tasks:
+        if tasks_to_drain:
+            asyncio.get_event_loop().run_until_complete(asyncio.wait([
+                asyncio.ensure_future(drain_and_kill_if_draining_fails(task, client)) for task, client in tasks_to_drain
+            ]))
```

Comment:
[nit] Totally understand why doing this on one line, but this is a loooong line, impairing code readability (same to the code segment on lines 205-208).

Reply:
Yeah, it's unfortunate. I've never found a way I like of splitting list comprehensions onto multiple lines. In a previous commit, I had done it by appending to an empty list, and I think this way is slightly more readable.